streams/streams.md

```diff
@@ -0,0 +1,126 @@
+# Processing large files with Node Streams
+
+A client of ours has a big Postgresql database that needs to be updated each day. These updates take place via incremental SQL files, provided on a daily basis. The incremental files are placed in a specific folder on the server. Our task is to apply these incremental files in sequence.
+
+## The incremental files
+Each incremental file starts with the following comments:
+
+```
+-- Increment timestamp: 20160129_192339
+-- Previous timestamp: 20160128_192500
+```
+
+Before we start applying these incremental files, we'd like to validate whether the previous timestamp of the first processed file matches the timestamp that's stored in the database. After that, we'd like to validate that the timestamps of all incremental files match with each other. We handle this validation-process with our existing Node-application.
+
+## Initial approach
+Our initial approach of handling this is by reading all files in the `incrementals` folder. Since filenames contain a date, sorting the files alphabetically within the folder gives us all available incremental files in order. For each file, we read it, abstract the increment and previous timestamps and validate whether the timestamps are in sequence.
+
+The (simplified) code for doing this:
+```js
+const fs = require('fs');
+const async = require('async');
+const path = require('path');
+
+let latest = '20160128_192500';
+let validFiles = [];
+
+// Read all files in ./incrementals folder
+fs.readdir('./incrementals', (e, files) => {
+  async.eachSeries(files.sort(), (file, cb) => {
+    fs.readFile(path.join('incrementals', file), (err, data) => {
+      const lines = data.toString().split('\r\n');
+
+      let previous = lines[1].split(' ').pop();
+      if (latest && latest !== previous) {
+        cb();
+        return;
+      }
+      latest = lines[0].split(' ').pop();
+      validFiles.push(file);
```

Comment:
`.filterSeries` is actually a better fit here since we might ignore files that are not in sequence